src/devices/multipleAnalogSensorsRosPublishers/GenericSensorRosPublisher.h

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2006-2019 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+
+#ifndef YARP_DEV_GENERICSENSORSROSPUBLISHER_H
+#define YARP_DEV_GENERICSENSORSROSPUBLISHER_H
+
+#include <yarp/os/PeriodicThread.h>
+#include <yarp/os/Publisher.h>
+#include <yarp/os/Node.h>
+#include <yarp/dev/DeviceDriver.h>
+#include <yarp/dev/IMultipleWrapper.h>
+#include <yarp/dev/MultipleAnalogSensorsInterfaces.h>
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+/**
+ * @ingroup dev_impl_wrapper
+ *
+ * \brief This abstract template needs to be specialized in a ROS Publisher, for a specific ROS mesagge/sensor type.
+ *
+ * | YARP device name |
+ * |:-----------------:|
+ * | `GenericSensorRosPublisher` |
+ *
+ * The parameters accepted by this device are:
+ * | Parameter name | SubParameter   | Type    | Units          | Default Value | Required                    | Description                                                       | Notes |
+ * |:--------------:|:--------------:|:-------:|:--------------:|:-------------:|:--------------------------: |:-----------------------------------------------------------------:|:-----:|
+ * | name           |      -         | string  | -              |   -           | Yes                         | Prefix of the port opened by this device                          | MUST start with a '/' character |
```

Comment:
Looking into the device it seems that this writes to the `${name}_publisher`, published by the `${name}_node` . As I guess most users do not care about the node name, while they may want to be able to explicitly set the used topic name (for example for compatibility with existing ROS nodes), perhaps it could make sense to change `name` to specify the topic, and then use `${name}_node` for the node name?

Reply:
It makes perfectly sense to me 👍